### PR TITLE
Add PEM fuzzer

### PR DESCRIPTION
**Pull request by kroeckx**
This fuzzer can find CVE-2022-4450

### CODE REVIEW

**Comment by kroeckx** [comment]
I will wait for #19669 before merging this

**Comment by kroeckx** [comment]
https://github.com/openssl/fuzz-corpora/pull/10 should preferably get merged before this.

File: fuzz/pem.c [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    BIO *in;
+    char *name = NULL, *header = NULL;
+    unsigned char *data = NULL;
+    long outlen;
+
+    if (len <= 1)
+        return 0;
+
+    in = BIO_new(BIO_s_mem());
+    OPENSSL_assert((size_t)BIO_write(in, buf+1, len-1) == len-1);
```
Comment (mattcaswell) [inline fuzz/pem.c:34]: Spaces around `+` and `-`

**Review by mattcaswell — APPROVED** [review]
Approved subject to the minor style issue being fixed.

**Review by mattcaswell — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Merged.

**Pull request closed without merging**